Deleting Decks and Cards

zsli16 opened the pull request:
Hey @carlhueffmeier! We added functions in the controllers to allow the user to delete a card and a deck. Please review the code and merge it if it's OK, so the app can use it in the client. Let us know if any changes are needed. Thanks! :)

carlhueffmeier commented on client/.env.example:
```diff
@@ -1 +1 @@
-REACT_APP_APOLLO_URL=URL of the server (*required)
\ No newline at end of file
+REACT_APP_APOLLO_URL=
```
Why are you removing documentation? 🙂

carlhueffmeier commented on client/src/components/NoteTypeSelect.js:
```diff
@@ -39,6 +39,7 @@ class NoteTypeSelect extends Component {
               if (error) {
                 return <p>Error! {error.message}</p>;
               }
+              console.log('allnoteTypes', allNoteTypes);
```
Never leave `console.log` in PR. Your colleagues are going to make fun of you 😉

carlhueffmeier commented on client/src/components/Signup.js:
```diff
@@ -19,10 +19,14 @@ const SIGNUP_MUTATION = gql`
 
+
 class Signup extends Component {
+  
```
You really like your whitespace, huh?

carlhueffmeier commented on client/src/components/DeckCreateForm.js:
```diff
@@ -14,6 +14,7 @@ const CREATE_DECK_MUTATION = gql`
 `;
 
 class DeckCreateForm extends Component {
+  
```
When making a PR, try avoiding committing unnecessary formatting changes. It blows up the diffs and creates noise in the git history. It also makes later git annotates or git blames more difficult (When trying to find out who is responsible for a line of code)

carlhueffmeier commented on server/app.js:
```diff
@@ -20,6 +20,7 @@ app.use(cookieParser());
 // Extract the user id and store for this request
 app.use((req, res, next) => {
   const userId = getUserIdFromCookie(req);
+  // const userId = 'Sues User Id';
```
😊...

carlhueffmeier commented on server/db/connectors/card.js:
```diff
@@ -122,6 +122,13 @@ function generateCardModel({ user } = {}) {
   };
 
+  const deleteOne = async (data = {}) => {
+    console.log('model reached', data)
```
🐥

carlhueffmeier commented on server/db/connectors/note.js:
```diff
@@ -16,6 +16,12 @@ function generateNoteModel({ user } = {}) {
 
+  const deleteOne = (where = {}) => {
+    checkAuth(user);
+    console.log('delete function in controller reached!');
```
🐥🐥

carlhueffmeier commented on server/graphql/entities/Card.js:
```diff
@@ -51,6 +56,12 @@ const getCard = (_, { where }, { db }) => {
 };
 
+const deleteCard = (_, { data }, { db }) => {
+  console.log('graphql reached', data.id)
```
🐥🐥🐥

carlhueffmeier commented on server/graphql/entities/Card.js:
```diff
@@ -51,6 +56,12 @@ const getCard = (_, { where }, { db }) => {
+const deleteCard = (_, { data }, { db }) => {
+  console.log('graphql reached', data.id)
+  const id = data.id
+  return db.card.deleteOne({id});
```
```suggestion
  return db.card.deleteOne({ id });
```
Consistency shows that you are a pro 😎

carlhueffmeier commented on server/graphql/entities/Card.js:
```diff
@@ -66,6 +77,11 @@ exports.resolvers = {
+    deleteCard
+  },
+
+  // Card: {
```
Commented out code.. 😅

carlhueffmeier commented on server/helper/utils.js:
```diff
@@ -30,7 +30,9 @@ const storeUserInCookie = (user, res) => {
 
 const getUserIdFromCookie = req => {
   const { token } = req.cookies;
+  console.log(token);
```
🐥🐥🐥🐥🐥🐥🐥🐥🐥

carlhueffmeier commented on server/graphql/entities/User.js:
```diff
@@ -60,6 +60,8 @@ const me = (_, __, { req }) => {
 
 const signup = async (_, { data: { email, password, name } }, { db, res }) => {
   // Create new user
+  console.log(email, password, name);
```
🐥🐥🐥🐥🐥

carlhueffmeier commented on server/graphql/entities/Note.js:
```diff
@@ -58,6 +64,12 @@ const updateNote = (_, { data }, { db }) => {
 
+const deleteNote = (_, { data }, { db }) => {
+  const { id } = data;
+  console.log('delete function in graphql reached!', id);
```
🐥🐥🐥🐥

carlhueffmeier commented on server/graphql/entities/Note.js:
```diff
@@ -58,6 +64,12 @@ const updateNote = (_, { data }, { db }) => {
+const deleteNote = (_, { data }, { db }) => {
+  const { id } = data;
+  console.log('delete function in graphql reached!', id);
+  return db.note.deleteOne({id});
```
```suggestion
  return db.note.deleteOne({ id });
```

carlhueffmeier commented on server/graphql/entities/Note.js:
```diff
@@ -34,6 +35,11 @@ exports.typeDef = gql`
 
+  input NoteDeleteInput {
+    id: ID!
+    deck: ID
```
Do you really need the deck to delete a note?

zsli16 commented:
Hey Carl! We are going to close this pull request and open a new one with the requested changes.

zsli16 commented:
Thanks for your feedback, Carl! We cleaned up the extra spaces and console logs.
We also removed the functions to delete a note, since we are only using the delete Card functionality on the frontend. The deck ID is not needed. Please let us know if any other changes are needed.
